generated/nidaqmx/_time.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import pytz
+from tzlocal import get_localzone
+from datetime import timezone
+from datetime import datetime as std_datetime
+from hightime import datetime as ht_datetime
+from typing import Optional, Union
+
+# theoretically the same as astimezone(), but with support for dates before 1970
+def _convert_to_desired_timezone(expected_time_utc: Union[std_datetime, ht_datetime], tzinfo: Optional[timezone] = None):
+    # if timezone matches, no need to do conversion
+    if expected_time_utc.tzinfo is tzinfo:
+        return expected_time_utc
+
+    # if timezone is not defined, use system timezone
+    if tzinfo is None:
+        local_timezone = get_localzone()
+        tzinfo = pytz.timezone(str(local_timezone))
+
+    # use pytz here to account for daylight savings
+    if isinstance(tzinfo, pytz.tzinfo.BaseTzInfo):
+        target_tz = pytz.timezone(str(tzinfo))
+        # we need to make expected_time_utc naive in order to use localize
+        expected_time_utc = expected_time_utc.replace(tzinfo=None)
+        localized_time = target_tz.localize(expected_time_utc)
+        desired_expected_time = target_tz.fromutc(localized_time)
```

Comment:
`pytz`'s localize method seems to call `dt.replace(tz=self)`: https://github.com/stub42/pytz/blob/master/src/pytz/__init__.py#L242

Do you really need `pytz`?